src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -515,43 +516,47 @@ public static CSharpCodeWritingScope BuildClassDeclaration(
             writer.WriteLine("#nullable restore");
         }
 
-        for (var i = 0; i < modifiers.Count; i++)
+        foreach (var modifier in modifiers)
         {
-            writer.Write(modifiers[i]);
+            writer.Write(modifier);
             writer.Write(" ");
         }
 
         writer.Write("class ");
         writer.Write(name);
 
-        if (typeParameters != null && typeParameters.Count > 0)
+        if (!typeParameters.IsDefaultOrEmpty)
         {
             writer.Write("<");
 
-            for (var i = 0; i < typeParameters.Count; i++)
+            var first = true;
+
+            foreach (var typeParameter in typeParameters)
```

Comment:
>foreach (var typeParameter in typeParameters) [](http://example.com/codeflow?start=12&length=45)

Could this use the has* technique used later in the method to not need the first local or check per loop iteration?

Reply:
redacted. Ignore me. :)